server/pages/index.js

```diff
@@ -376,20 +378,44 @@ module.exports = function() {
 		app.get( '/themes/:theme_slug', function( req, res ) {
 			const context = getDefaultContext( req );
 
-			if ( config.isEnabled( 'server-side-rendering' ) ) {
+			const renderThemeSheet = function( theme ) {
 				const store = createReduxStore();
+				store.dispatch( {
+					type: ActionTypes.RECEIVE_THEME_DETAILS,
+					themeId: theme.id,
+					themeName: theme.name,
+					themeAuthor: theme.author,
+					themeScreenshot: theme.screenshot,
+				} );
 
 				store.dispatch( setSection( 'themes', { hasSidebar: false, isFullScreen: true } ) );
-				context.initialReduxState = pick( store.getState(), 'ui' );
+				context.initialReduxState = pick( store.getState(), 'ui', 'themes' );
 
 				Object.assign( context, render( (
 					<ReduxProvider store={ store }>
 						<LayoutLoggedOutDesign store={ store } routeName={ 'themes' } match={ { theme_slug: req.params.theme_slug } } />
 					</ReduxProvider>
 				) ) );
-			}
+				res.render( 'index.jade', context );
+			};
 
-			res.render( 'index.jade', context );
+			if ( config.isEnabled( 'server-side-rendering' ) ) {
+				let themeData = themeDetails.get( req.params.theme_slug );
+				if ( ! themeData ) {
+					wpcom.undocumented().themeDetails( req.params.theme_slug, ( error, data ) => {
```

Comment:
> I wonder if we could get away without having to use memcache by...

Yeah, was thinking about this, or maybe even using the api response header, which may let us leverage some existing libraries to handle it?

Once we receive new api data, how do we clear the renderToString cache?